src/nvim/CMakeLists.txt

```diff
@@ -221,15 +221,33 @@ function(get_preproc_output varname iname)
   endif()
 endfunction()
 
-set(NVIM_VERSION_GIT_H ${PROJECT_BINARY_DIR}/cmake.config/auto/versiondef_git.h)
+execute_process(
+  COMMAND git rev-parse --is-inside-work-tree
+  OUTPUT_VARIABLE IS_INSIDE_GIT_TREE
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+
+if(IS_INSIDE_GIT_TREE)
```

Comment:
no need for this. `GenerateVersion.cmake` can just produce an empty file, or something like that, if `git` fails.